pkg/kubelet/certificate/bootstrap/bootstrap.go

```diff
@@ -98,23 +139,47 @@ func LoadClientCert(kubeconfigPath string, bootstrapPath string, certDir string,
 		}
 	}
 
-	if err := waitForServer(*bootstrapClientConfig, 1*time.Minute); err != nil {
-		klog.Warningf("Error waiting for apiserver to come up: %v", err)
+	interval := wait.Backoff{
+		Duration: time.Second,
+		Factor:   1.1,
+		Steps:    20,
+		Jitter:   0.2,
+		Cap:      30 * time.Second,
 	}
+	for interval.Steps > 0 {
+		// creating a new client with a new Dial function ensures we get a new cache entry
+		// TODO: it also leaks cache entries, which is a broader problem
```

Comment:
https://github.com/kubernetes/kubernetes/pull/58677 would help here.

We need to make sure transports are also cleaned up. I don't think that will happen properly when bootstrapClient is GC'd:

https://github.com/golang/go/issues/24739

Reply:
It won't, which is why Jordan suggested a "bypass reuse" option.  I don't think the TLS cache holding things forever is a reasonable long term path, but I could be persuaded it's a short term avoidable problem.

I could just create a separate transport here and bypass the cache completely, which may just be the simplest option.